### PR TITLE
 Closes #322: Refactor dataset summary 

### LINKED ISSUE #322: Try to remove field values extraction in DatasetVersionUI and display summary fields using only getDatasetSummaryFields

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetFieldUtil.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * Utility class with common operations on dataset fields
+ * 
+ * @author madryk
+ */
+public class DatasetFieldUtil {
+
+    private DatasetFieldUtil() {
+        throw new IllegalArgumentException("Could not be instantiated");
```
Comment (MrK191) [inline dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetFieldUtil.java:18]: Is it necessary? I think programmers will know after checking out the class.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetFieldsInitializer.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author skraffmiller
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetFieldsInitializer.java]: I think we can remove it :>

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetFieldUtil.java (hunk 1 above)
Comment (madryk) [reply dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetFieldUtil.java:18]: It won't hurt. This is some kind of additional protection. Without this it is still possible to create instance of this class using reflection

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetFieldsInitializer.java (hunk 2 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetFieldsInitializer.java]: ok

**Review by MrK191 — APPROVED** [review]

**Pull request merged**